test/public/runs/detail.test.js

```diff
@@ -527,4 +553,29 @@ module.exports = () => {
             innerText: 'CmCvjNbg',
         });
     });
+
+    it('should display correct tag styling after updating in tag overview', async () => {
+        let badges;
+        // Fetch the run data before update of tag
+        await goToRunDetails(page, 106);
+
+        badges = await getBadges();
+
+        expect(badges[0].backgroundColor === 'rgb(255, 0, 0)').to.be.false;
+
+        const updatedTag = {
+            color: '#ff0000', // Red color
+        };
+
+        // Update the tag via API request
+        await request(server)
+            .put('/api/tags/1?token=admin')
+            .send(updatedTag)
+            .expect(201);
```

Comment:
Yes that works, but I think you can make it more maintainable by directly calling the UpdateTagUseCase to avoid relying on the API.

Reply:
Done